dojo/models.py

```diff
@@ -1528,7 +1528,7 @@ class Meta:
 
 class Test(models.Model):
     engagement = models.ForeignKey(Engagement, editable=False, on_delete=models.CASCADE)
-    lead = models.ForeignKey(User, editable=True, null=True, on_delete=models.RESTRICT)
+    lead = models.ForeignKey(User, editable=True, null=True, blank=True, on_delete=models.SET_NULL)
```

Comment:
Why have you changed `on_delete=models.RESTRICT` with `on_delete=models.SET_NULL`? I don't think it's a good idea to change the lead in tests when deleting a user. When somebody wants to delete a user, they should look at all occurences of the user. That's why all ForeignKeys with `Dojo_User` or `User` are restricted.

Reply:
For now, the code of the importer/re-importer allow to have lead to be `None`. There is also no input for it in the UI, and in the API v2 you can push a report without a `lead` value. So for me it's better to consider it optional in the model to be aligned with the rest of the code and make it possible to remove the user without need to update every Test/engagement where the user was a lead. But I can change it. I don't think it's critical to this patch.

Reply:
It's fine for the `lead` to be optional. But please change it back to restricted, makes it consistent with the other occurences of users.

Reply:
done